src/Relations/MorphToMany.php

```diff
@@ -185,15 +213,19 @@ public function attach($id, array $attributes = [], $touch = true)
 
             if ($this->getInverse()) {
                 // Attach the new ids to the parent model.
-                $this->parent->push($this->table, [
-                    [
-                        $this->relatedPivotKey => $model->{$this->relatedKey},
-                        $this->morphType => $model->getMorphClass(),
-                    ],
-                ], true);
+                if ($this->parent instanceof \MongoDB\Laravel\Eloquent\Model) {
+                    $this->parent->push($this->table, [
+                        [
+                            $this->relatedPivotKey => $model->{$this->relatedKey},
+                            $this->morphType => $model->getMorphClass(),
+                        ],
+                    ], true);
+                } else {
+                    $this->addIdToParentRelationData($id);
+                }
 
                 // Attach the new parent id to the related model.
-                $model->push($this->foreignPivotKey, $this->parseIds($this->parent), true);
+                $model->push($this->foreignPivotKey, (array) $this->parent->{$this->parentKey}, true);
```

Comment:
`(array)` cast should not be used. If the value is an `ObjectId`, the result is not the one you expect.

```php
var_dump((array) new MongoDB\BSON\ObjectId());
/*
array(1) {
  'oid' =>
  string(24) "6577071f3fded0903a01f030"
}
*/
```

Here we know it's not a list of ids, so a simple array wrapping is fine.
```suggestion
                $model->push($this->foreignPivotKey, [$this->parent->{$this->parentKey}], true);
```

Reply:
I will double-check it later. thanks for the review❤️

Reply:
The `ObjectId` fields are always `string`. I tested this with a primary key and a custom key and a custom key was cast to `ObjectId`.
In all cases, the `$this->parent->{$this->parentKey}` value is `string`. Your suggestion is more readable, but I don't see any bug here to open a PR.

Reply:
Yes, that's something we'll have to fix if we manage to store references as ObjectId. But there are many other hurdles to overcome, and I think this notion of "all keys are strings" is too hard-wired into Eloquent anyway.